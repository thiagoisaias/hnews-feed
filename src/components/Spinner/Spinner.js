import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StyledSpinner = styled.svg`
  animation: rotate 2s linear infinite;
  width: ${props => props.width || "24px"};
  height: ${props => props.height || "24px"};

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }

  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -125;
    }
  }
`;

const StyledCircle = styled.circle`
  stroke: #999;
  stroke-linecap: round;
  stroke-width: 2.5;
  animation: dash 1.5s ease-in-out infinite;
`;

const Spinner = props => {
  const { width, height } = props;
  return (
    <Container>
      <StyledSpinner width={width} height={height} viewBox={`0 0 ${width*2} ${height*2}`}>
        <StyledCircle cx={width} cy={height} r={width / 2} fill="none" />
      </StyledSpinner>
    </Container>
  );
};

Spinner.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired
};

export default Spinner;
