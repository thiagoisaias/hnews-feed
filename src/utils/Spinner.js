import React from "react";
import styled from "styled-components";

const Container = styled.div`
  width: 50px;
  height: 50px;
  margin: 32px auto;
`;

const StyledSpinner = styled.svg`
  animation: rotate 2s linear infinite;
  width: 50px;
  height: 50px;

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }

  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -125;
    }
  }
`;

const StyledCircle = styled.circle`
  stroke: #777;
  stroke-linecap: round;
  stroke-width: 1.725;
  animation: dash 1.5s ease-in-out infinite;
`;

const Spinner = props => {
  return (
    <Container>
      <StyledSpinner viewBox="0 0 50 50">
        <StyledCircle cx="25" cy="25" r="15" fill="none" />
      </StyledSpinner>
    </Container>
  );
};

export default Spinner;
