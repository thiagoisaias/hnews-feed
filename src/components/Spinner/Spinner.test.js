import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Spinner from "./Spinner";

describe("Spinner", () => {
  it("renders properly", () => {
    const component = shallow(<Spinner width={32} height={32} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
