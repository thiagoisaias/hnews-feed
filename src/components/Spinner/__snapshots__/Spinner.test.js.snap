// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders properly 1`] = `
<styled.div>
  <styled.svg
    height={32}
    viewBox="0 0 64 64"
    width={32}
  >
    <styled.circle
      cx={32}
      cy={32}
      fill="none"
      r={16}
    />
  </styled.svg>
</styled.div>
`;
